### ISSUE TITLE
Can this client support NTFS Streams?

### ISSUE THREAD

**Issue by SRGOM** [abraunegg/onedrive#780]
**Is your feature request related to a problem? Please describe.**
Not exactly a problem because nobody really uses them but if somebody does, then probably yeah... 

### About streams:
You might have to read up a tiny bit on NTFS Streams but they are alternative data streams, accessed using a `:` after filename. Say a file has 3 streams: myfile, myfile:not_so_secret_data1, myfile:not_so_secret_data2. 

### How I came across
I started an instance of this client and compared it against an old Windows mount (which had recent OneDrive data). Apparently the NTFS driver on Linux is good enough to list streams as separate `:` having files so `diff -r` revealed the difference. I had uploaded the file multiple computers back so my guess is that OneDrive cloud service and Windows OneDrive client both recognize these streamss

### How would one reproduce this
Here's a powershell command to read and write:

```
Set-Content -Path d:\File1 -Value "Hello, World" -Stream Stream1
Get-Content -Path d:\File1 -Stream Stream1
```
I would test this using the following way:

1. Create a new file with an alternate data stream, on an NTFS partition. (I believe the latest ReFS on servers also supports these)

2. Copy paste to OneDrive location (should also be NTFS)

3. Let it upload

4. Download to a new OneDrive instance (again, NTFS).

5. Check using powershell if stream exists. 

6. If not, then do nothing since at least one link in the official chain is broken.  If yes, then do something :). (Solution below)

**Describe the solution you'd like**
What's happening right now is seemingly the ADS (alternate data stream) not getting downloaded. I believe it's safe to download it with a colon in name. So the above file would be named File1:Stream1. An upload should break up the string and appropriately handle the names.

**Additional context**
Truncated output of `diff -r`

```
Only in /mnt/partitions/sdb11/OneDrive/Documents/Archive/MSDN/elms File：elms_support@e-academy.com - (Software Receipt and Installation Instructions).eml:OECustomProperty:$DATA
Only in /mnt/partitions/sdb11/OneDrive/Documents/Archive/MSDN/elms File：system_mail@e-academy.com - (Welcome to MSDN Academic Alliance Online Software System).eml:OECustomProperty:$DATA
```

Very happy to add more details...

**Comment by abraunegg** [comment]
@SRGOM 
Will do some reading, however at first glance:

> accessed using a : after filename.

: character is in violation of https://docs.microsoft.com/en-au/windows/win32/fileio/naming-a-file?redirectedfrom=MSDN ... so if the file is access by that, and it is coupled to say on-demand download or even to sync, the properties of the file with : in it would be rejected by the OneDrive API from syncing.

What I will do however is as you have suggested test this on 2 x Windows systems to see what the official client does in respect to these types of files

**Comment by SRGOM** [comment]
Please ignore what I said in the 'about streams' section. I don't remember where I read it, perhaps somewhere in the WSL discussion but that's why colon isn't alowed in a file name itself. I seem to recall having done a "file:stream" read using probably a rust program but again, please ignore. The powershell command definitely works.

Also, the Linux ntfs driver, at least on my system, makes the resultant stream a full file named file:stream

**Comment by abraunegg** [comment]
@SRGOM 
I have tested with an up-to-date Windows 10 platform and it appears that 'streams' are **not** uploaded / transferred by OneDrive:

**Windows 10 - System 1 - User 1**
![system1](https://user-images.githubusercontent.com/4956234/73044838-9e13e100-3ebf-11ea-9a01-5b52e950ae8e.jpg)

**Windows 10 - System 2 - User 2**
![system2](https://user-images.githubusercontent.com/4956234/73044850-a8ce7600-3ebf-11ea-9902-9ecbebaae19b.jpg)


So based on this, nothing more to see or do - OneDrive does not support 'streams' based on this testing. If this is not correct, please can you provide a dummy file with a stream via an archive file & email for testing.

**Comment by SRGOM** [comment]
I don't own a Windows License right now unfortunately. 

That said, stream information is lost in most operations such as archiving/emailing. What I do have is a file (info at the very end of my original comment) which the linux NTFS driver has split into two files, one with a colon name. This file was downloaded using OneDrive. The question now becomes how to upload such a file. My guess is OneDrive's latest clients do not upload stream information (since I did manage to download earlier). I will list it in my ToDo and get back to you when I can (which may not be soon). Closing this for now but will reopen later.
